### PR TITLE
Add a random wait before restarting

### PR DESCRIPTION
**Pull request by iwiznia**
Tests:
- Started BWM
- Touched the restart file
- Check it logs 
```
2022-05-06T15:44:01.333973+00:00 expensidev2004 php: m7KoMw vendor/bin/BedrockWorkerManager.php we@dont.know !script! ?api? [info] Version watch file changed, will stop processing new jobs soon ~~ wait: '53'
```
- Wait a bit, check it logs:
```
[2022-05-06T15:44:57.579356+00:00 expensidev2004 php: m7KoMw vendor/bin/BedrockWorkerManager.php we@dont.know !script! ?api? [info] Version watch file changed and wait time is over, stop processing new jobs
2022-05-06T15:44:57.579646+00:00 expensidev2004 php: m7KoMw vendor/bin/BedrockWorkerManager.php we@dont.know !script! ?api? [info] Version watch file changed, stop processing new jobs
2022-05-06T15:44:57.579740+00:00 expensidev2004 php: m7KoMw vendor/bin/BedrockWorkerManager.php we@dont.know !script! ?api? [info] Stopping BedrockWorkerManager, waiting for children
```

### CODE REVIEW

File: bin/BedrockWorkerManager.php [hunk 1]
```diff
@@ -637,15 +638,24 @@ function getNumberOfJobsToQueue(): int
  *
  * @param Expensify\Bedrock\Stats\StatsInterface $stats
  */
-function checkVersionFile(string $versionWatchFile, int $versionWatchFileTimestamp, $stats): bool
+function checkVersionFile(string $versionWatchFile, int $versionWatchFileTimestamp, LoggerInterface $logger): bool
 {
-    return $stats->benchmark('bedrockWorkerManager.checkVersionFile', function () use ($versionWatchFile, $versionWatchFileTimestamp) {
```
Comment (johnmlee101) [inline bin/BedrockWorkerManager.php:642]: Did we ever use this stat logging lol

Comment (iwiznia) [reply bin/BedrockWorkerManager.php:642]: No it is pretty useless and was giving me headaches on how to use the static variable inside the anonymus function, so removed it.

**Comment by johnmlee101** [comment]
I'm guessing this is on hold because we might close this out then?

**Comment by iwiznia** [comment]
Yeah

**Comment by iwiznia** [comment]
I pushed here the code for waiting for child jobs that have retryAfter only, just so it's saved, but will open a new PR with the changes mentioned [here](https://github.com/Expensify/Expensify/issues/209619#issuecomment-1121294040)

**Pull request closed without merging**